### PR TITLE
fix(extractors): align HCL/Dart/Scala contains parity across engines

### LINKED ISSUE #1189: tracking: cross-engine `contains`/`parameter_of` parity for 11 language extractors
## Summary

After fixing the JavaScript callback-callee gating (#1191), the v3.10.1-dev.80 dogfood report's 36% `receiver` / 5.5% `calls` gap drops to ~0% on JS/TS/TSX. The residual `contains` (-51) and `parameter_of` (-35) deltas from that report come from **per-language divergences across 11 extractors**, captured in this tracking issue.

The C++ reference-parameter bug (#1187) and three C# divergences (#1188) are tracked separately because each has a single concrete fix. This issue covers the remaining language-by-language deltas that need individual investigation.

## Surface area

Edges that exist in WASM but not native, **or vice versa**, broken down by source-file extension on the codegraph repo build (`--no-incremental`, all WASM grammars present):

| Language | WASM-only `contains` | Native-only `contains` | Pattern (TBD per lang) |
|----------|---------------------:|-----------------------:|------------------------|
| Elixir (`.ex`)       | 24 | 0  | likely module body extraction |
| Haskell (`.hs`)      | 24 | 0  | likely data/where clause |
| Python (`.py`)       | 13 | 0  | possibly decorators or nested def |
| Terraform (`.tf`)    | 12 | 0  | block/resource body items |
| Dart (`.dart`)       | 10 | 0  | TBD |
| Ruby (`.rb`)         | 1  | 0  | TBD |
| Kotlin (`.kt`)       | 1  | 8  | both directions — different patterns |
| Objective-C (`.m`)   | 1  | 1  | both directions |
| CUDA (`.cuh`)        | 2  | 2  | both directions |
| Scala (`.scala`)     | 0  | 11 | native-only — native extracts more |
| Java (`.java`)       | 0  | 4  | native-only |

`parameter_of` mirrors `contains` for the same kinds (parameter ↔ function), so the residual `parameter_of -35` decomposes similarly into the same languages.

## Goal

For each language above, determine the root cause of the divergence and decide which engine is correct, then align. Per CLAUDE.md, divergence = bug in the less-accurate engine, not an acceptable parity gap.

## Suggested approach

1. For each language, repro on `tests/benchmarks/resolution/fixtures/<lang>/` and inspect the WASM-only / native-only edges (`source/target name+kind+line` triples) to identify the divergent AST pattern.
2. Decide correct behavior per language, fix the wrong engine.
3. Add the fixture pattern to `tests/engines/parity.test.ts` as a regression guard.

## Out of scope

- JavaScript/TypeScript callback-callee gating — fixed in #1191
- C++ reference-parameter naming — tracked in #1187
- C# auto-property kind / implements / interface-param — tracked in #1188

## Why one tracking issue

Each language needs its own investigation, but the pattern (diff fixture → identify divergent AST handling → fix one extractor) is identical. Keeping these in one ticket lets reviewers see the audit shrink as languages get fixed, and avoids 11 near-identical issues.

### PR DESCRIPTION
**Pull request by carlos-alm**
## Summary

Three more per-language divergences from the v3.10.1-dev.80 dogfood report's residual `contains` gap, refs #1189.

- **HCL native** — `handle_hcl_block` emitted `variable`/`output` definitions without attribute children → added `extract_block_attributes` that pulls each attribute key as a `property` child (mirrors WASM `extractHclAttributes`). 12 missing contains edges in the fixture.
- **Dart native** — `extract_dart_class_methods` iterated `class_body` children looking for `method_signature`/`function_signature` directly, but `tree-sitter-dart 0.0.4` (the version pinned by the Rust crate) wraps each signature in a `class_member_definition`. The fix unwraps that container before matching. 10 missing contains edges in the fixture.
- **Scala WASM** — `extractScalaBodyMembers` emitted class/trait/object methods with no parameter children, while native correctly extracted them via `handle_scala_function_definition`. WASM now passes the parameter list as `children`, matching native. 11 missing contains edges in the fixture.
- **Scala WASM emission order** also realigned: the class/trait/object handlers previously pushed methods *before* the parent type, while native pushes the parent first (source order). Refactored through `emitScalaTypeDef` + `collectScalaBodyMembers` so WASM matches native's depth-first source-line order — required for the new parity assertion to pass.

## Verification

- `npx vitest run tests/engines/parity.test.ts` — 13 passed, 3 skipped (HCL case un-skipped; Dart/Scala cases newly added).
- `npx vitest run tests/parsers/ tests/engines/` — 567 passed.
- `npx vitest run tests/integration/ tests/graph/ tests/benchmarks/resolution/` — 971 passed.
- Cross-engine contains-edge diff on `tests/benchmarks/resolution/fixtures/{hcl,dart,scala}/*` is now zero in both directions.

## Test plan

- [x] HCL `variable`/`output` blocks produce `property` children in both engines.
- [x] Dart class methods extracted by native via the `class_member_definition` unwrap path.
- [x] Scala class/trait/object method `parameter` children present in both engines.
- [x] Scala definition emission order matches native (parent before members).
- [x] All existing parity, parser, integration, and resolution tests pass.

Closes #1189 partially (HCL, Dart, Scala). Remaining: Java, Kotlin, Objective-C, CUDA, Ruby (single-edge case).

### CODE REVIEW

File: crates/codegraph-core/src/extractors/dart.rs [hunk 1]
```diff
@@ -94,17 +94,26 @@ fn handle_dart_class(node: &Node, source: &[u8], symbols: &mut FileSymbols) {
...
-            match member.kind() {
+            // tree-sitter-dart 0.0.4 wraps method/function signatures in
+            // `class_member_definition` (a single-child container). Drill in
+            // before matching; newer grammar versions place the signatures
+            // directly under `class_body`.
+            let sig = if member.kind() == "class_member_definition" {
+                member.child(0).unwrap_or(member)
+            } else {
+                member
+            };
```
[the remark this answers is not in the document]
Comment (carlos-alm) [reply crates/codegraph-core/src/extractors/dart.rs:110]: Fixed in 36f113b. Replaced the absolute `child(0)` with a search for the first `method_signature` / `function_signature` child of `class_member_definition` via a new `find_dart_signature_child` helper. Any leading anonymous, metadata, or modifier nodes (now or in a future grammar revision) are skipped over instead of silently dropping the method.

File: tests/engines/parity.test.ts [hunk 2]
```diff
@@ -287,16 +276,45 @@ class Controller {
...
+      // emitted the method definition without children, while native did.
+      name: 'Scala — method parameters as children of class methods',
+      file: 'Svc.scala',
+      code: `
+class UserService {
+  def createUser(id: String, name: String): String = name
+  def removeUser(id: String): Boolean = true
 }
 `,
     },
```
[the remark this answers is not in the document]
Comment (carlos-alm) [reply tests/engines/parity.test.ts:408]: Fixed in 7cd8191. Restored the original `HCL — resources and modules` test case un-skipped, sitting alongside the new variable/output attributes case so the `resource`/`module` extraction paths now have an active parity assertion.

**Comment by carlos-alm** [comment]
@greptileai

**Comment by carlos-alm** [comment]
Addressed the remaining Greptile suggestion in 9c5d50e: added an `object DefaultGreeter extends Greeter` parity case to `tests/engines/parity.test.ts` so the new inheritance-tracking path through `emitScalaTypeDef` for Scala objects is now exercised in both engines. Also resolved a merge conflict with main (PR #1195 landed) in 76c4328 — `tests/engines/parity.test.ts` now contains both the Elixir/Haskell and the HCL/Dart/Scala parity cases.

**Comment by carlos-alm** [comment]
@greptileai
